### PR TITLE
Upgrade Go version to 1.18 and replace io/ioutil references with io or os

### PR DESCRIPTION
**Pull request by vyasgun**
# Changes

- Upgraded Go to 1.18
- Replaced `io/ioutil` references in favour of `io` or `os`
- Upgrade knative.dev/eventing dependency to v0.35.0

/kind cleanup

Fixes #

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

File: .github/workflows/knative-boilerplate.yaml [hunk 1]
```diff
@@ -30,15 +30,15 @@ jobs:
       fail-fast: false # Keep running if one leg fails.
       matrix:
         extension:
-        - go
-        - sh
+          - go
```
Comment (dsimansk) [inline .github/workflows/knative-boilerplate.yaml:33]: Was it formatted with IDE maybe? It will be overriden by GH action sync anyway, just curious.

**Comment by dsimansk** [comment]
/hold

There's the following code style check. We should probably take a look on syncing existing code in client-pkg with latest from client.   

```
Running [/home/runner/golangci-lint-1.50.1-linux-amd64/golangci-lint run --out-format=github-actions --go=1.19] in [] ...
  Error: SA1019: table.SelfLink is deprecated: selfLink is a legacy read-only field that is no longer populated by the system. +optional  (staticcheck)
  Error: SA1019: table.SelfLink is deprecated: selfLink is a legacy read-only field that is no longer populated by the system. +optional  (staticcheck)
  
  Error: issues found
  Ran golangci-lint in 130636ms
```

File: .github/workflows/knative-boilerplate.yaml (hunk 1 above)
Comment (vyasgun) [reply .github/workflows/knative-boilerplate.yaml:33]: Yes, the IDE formatted it.

**Pull request merged**